website/pages/docs/upgrading/instructions/general-process.mdx

```diff
@@ -0,0 +1,176 @@
+binary with the new one.
+
+**3.** Perform a rolling restart of Consul on your servers, leaving the leader agent
+for last. Only restart one server at a time. After restarting each server, validate
+that it has rejoined the cluster and is in sync with the leader by issuing the `consul info` command,
+and checking whether the `commit_index` and `last_log_index` fields have the same value.
+If done properly, this should avoid an unexpected leadership election due to loss of quorum.
+
+~> It is important to issue a `consul leave` command on each server node when shutting
+Consul down. Make sure your service management system (e.g., systemd, upstart, etc.) is
+performing that action. If not, make sure you do it manually or you _will_ end up in a
+bad cluster state.
+
+**4.** Double-check that all servers are showing up in the cluster as expected and are on
+the correct version by issuing:
+
+```
+consul members
+```
+
+You should receive output similar to this:
+
+```
+Node       Address         Status  Type    Build  Protocol  DC
+dc1-node1  10.11.0.2:8301  alive   server  1.8.3  2         dc1
+dc1-node2  10.11.0.3:8301  alive   server  1.8.3  2         dc1
+dc1-node3  10.11.0.4:8301  alive   server  1.8.3  2         dc1
+```
+
+Also double-check the raft state to make sure there is a leader and sufficient voters:
+
+```
+consul operator raft list-peers
+```
+
+You should receive output similar to this:
+
+```
+Node       ID                                    Address         State     Voter  RaftProtocol
+dc1-node1  ae15858f-7f5f-4dcb-b7d5-710fdcdd2745  10.11.0.2:8300  leader    true   3
+dc1-node2  20e6be1b-f1cb-4aab-929f-f7d2d43d9a96  10.11.0.3:8300  follower  true   3
+dc1-node3  658c343b-8769-431f-a71a-236f9dbb17b3  10.11.0.4:8300  follower  true   3
+```
+
+**5.** Set your `log_level` back to what you had it at prior to the upgrade and issue
+`consul reload` again.
+
+## Troubleshooting
+
+Most problems with upgrading occur due to either failing to upgrade the leader agent last,
+or due to failing to wait for a follower agent to fully rejoin a cluster before moving
+on to another server. This can cause a loss of quorum and occasionally can result in
+all of your servers attempting to kick off leadership elections endlessly without ever
+reaching a quorum and electing a leader.
+
+Most of these problems can be solved by following the steps outlined in our
+[Outage Recovery](https://learn.hashicorp.com/tutorials/consul/recovery-outage) document.
+If you are still having trouble after trying the recovery steps outlined there,
+then these options for further assistance are available:
+
+- OSS users without paid support plans can request help in our [Community Forum](https://discuss.hashicorp.com/c/consul/29)
+- Enterprise and OSS users with paid support plans can contact [HashiCorp Support](https://support.hashicorp.com/)
+
+If you end up contacting support, please make sure you include the following information
```

Comment:
When contacting Hashicorp Support, please include the following information in your ticket: